GPUVerifyLib/GPUVerifyErrorReporter.cs

```diff
@@ -541,7 +541,7 @@ private static void PopulateModelWithStatesIfNecessary(Counterexample cex)
             }
         }
 
-        private static void DetermineNatureOfRace(CallCounterexample callCex, out string raceName, out string access1, out string access2)
+        protected static void DetermineNatureOfRace(CallCounterexample callCex, out string raceName, out string access1, out string access2)
```

Comment:
Instead of calling this method and then `AccessType.Create(access1)`, it would be better to expose `GetAccessType` and call `GetAccessType(callCex)`. Although you currently store the result of `DetermineNatureOfRace` on you side, you don't seem to do anything with the result, so it should be easy to remove.

Reply:
Leaving this method as private and making GetAccessType as protected.